examples/math/run_cleanup_webpages_with_llm.py

```diff
@@ -0,0 +1,235 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import glob
+import os
+from datetime import datetime
+
+import pandas as pd
+from loguru import logger
+
+from nemo_curator.backends.xenna import XennaExecutor
+from nemo_curator.core.client import RayClient
+from nemo_curator.models.vllm_model import _MODELS
+from nemo_curator.pipeline import Pipeline
+from nemo_curator.stages.math import LLMCleanupStage, TokenSplitterStage
+from nemo_curator.stages.resources import Resources
+from nemo_curator.stages.text.io.reader import JsonlReader, ParquetReader
+from nemo_curator.stages.text.io.writer import JsonlWriter
+from nemo_curator.stages.text.modules.modifier import Modify
+from nemo_curator.utils import prompts
+
+
+def fill_null_text(text: str | None) -> str:
+    """Fill null/NaN text values with empty string."""
+    if pd.isna(text) or text is None:
+        return ""
+    return str(text)
+
+
+def build_pipeline(  # noqa: PLR0913
+    input_files: list[str],
+    reader_type: str,
+    output_dir: str,
+    model: str,
+    prompt: str,
+    chunk_length: int | None = None,
+    chunk_data: bool = False,
+    classification: bool = False,
+    max_model_len: int | None = None,
+    temperature: float = 0.7,
+    top_p: float = 0.8,
+    top_k: int = 20,
+    min_p: float = 0.0,
+    max_tokens: int | None = None,
+    cache_dir: str | None = None,
+) -> Pipeline:
+    """Build the LLM cleanup pipeline."""
+    p = Pipeline(
+        name="math_cleanup_webpages_with_llm",
+        description="Clean up HTML/text content using LLM (with optional chunking)",
+    )
+
+    # Reader stage
+    if reader_type == "parquet":
+        p.add_stage(
+            ParquetReader(file_paths=input_files).with_(
+                {
+                    "file_partitioning": {"resources": Resources(cpus=0.5)},
+                    "parquet_reader": {"resources": Resources(cpus=0.5)},
+                }
+            )
+        )
+    else:
+        p.add_stage(
+            JsonlReader(file_paths=input_files).with_(
+                {
+                    "file_partitioning": {"resources": Resources(cpus=0.5)},
+                    "jsonl_reader": {"resources": Resources(cpus=0.5)},
+                }
+            )
+        )
+
+    p.add_stage(
+        Modify(modifier_fn=fill_null_text, input_fields="text", output_fields="text").with_(
+            resources=Resources(cpus=0.5)
+        )
+    )
```

Comment:
I am curious where some of these resource assignments are coming from. I belive `Modify` is probably better with `cpus=0.5`, did you set this because that's what you were observing?